mio-anonymous-pipes/src/lib.rs

```diff
@@ -0,0 +1,369 @@
+extern crate mio;
+extern crate miow;
+extern crate winapi;
+
+use mio::{Evented, Poll, PollOpt, Ready, Registration, SetReadiness, Token};
+use miow::pipe::{AnonRead, AnonWrite};
+
+use winapi::um::ioapiset::CancelSynchronousIo;
+
+use std::io;
+use std::os::windows::io::AsRawHandle;
+use std::sync::{Arc, atomic::{AtomicBool, Ordering}, Condvar, Mutex};
+use std::thread::{JoinHandle, spawn};
+
+/// Fixed-size ring buffer
+struct CursorBuffer {
+    // memory
+    buf: Box<[u8]>,
+
+    // start of initialised region
+    start: usize,
+    // length of initialised region (might wrap around to the start)
+    len: usize
```

Comment:
Some style issues with these comments and empty lines. There shouldn't be an empty line between `buf` and `start`. Also since `// Fixed-size ring buffer` is capitalized (and most other comments in the codebase are too iirc), the rest should also be.